opentracing-jdbc/opentracing-jdbc-autoconfigure/src/main/java/org/zalando/opentracing/jdbc/autoconfigure/TracingProcessor.java

```diff
@@ -1,13 +1,19 @@
 package org.zalando.opentracing.jdbc.autoconfigure;
 
+import org.springframework.beans.factory.BeanFactory;
 import org.zalando.opentracing.jdbc.DataSourceTracer;
 
 import javax.sql.DataSource;
 
 final class TracingProcessor extends ForwardingProcessor {
 
-    public TracingProcessor(final DataSourceTracer tracer) {
-        super(new ConditionalProxyProcessor<>(DataSource.class, tracer::trace));
+    public TracingProcessor(final BeanFactory beanFactory) {
```

Comment:
looking at other post processors, apparently the "idiomatic" way is to mark the processor as `BeanFactoryAware`, but I wanted to keep the changes minimal not break up `ConditionalProxyProcessor`.